File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if(x <= 1) return x;

        int left = 1, right = x/2;
        while(true){
            var mid = left + (right - left)/2;
            if(mid > x / mid)
                right = mid - 1;
            else if(mid+1 > x / (mid+1) )
                    return mid;
            else
                left = mid + 1;
        }
    }
}

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x <= 1: return x

        left, right = 1, x//2
        while True:
            mid = left + (right - left)//2
            if mid * mid > x:
                right = mid - 1
            elif (mid+1) * (mid+1) > x:
                    return mid
            else:
                left = mid + 1